modules/mod_ssl_self_signed/mod_ssl_self_signed.erl

```diff
@@ -90,15 +92,16 @@ check_keyfile(Filename) ->
 			{error, {cannot_read_pemfile, Filename, Error}}
 	end.
 
-check_dhfile(Filename) ->
+check_dhfile(Filename, Default) ->
     case filelib:is_file(Filename) of
         true ->
             case is_dhfile(Filename) of
                 true -> ok;
                 false -> {error, {no_dhfile, Filename}}
             end;
         false ->
-            generate_dhfile(Filename)
+            FileData = dh_params(Default),
```

Comment:
Need to map the default to an atom (or to a binary and then use that below), as the `m_config:get_value/3` will return a binary.

Reply:
Missed that one. I only tested it with a site config file.